app/javascript/dashboard/routes/dashboard/settings/reports/LiveReports.vue

```diff
@@ -108,6 +125,41 @@ export default {
     fetchAllData() {
       this.fetchAccountConversationMetric();
       this.fetchAgentConversationMetric();
+      this.fetchHeatmapData();
+    },
+    fetchHeatmapData() {
+      if (this.uiFlags.isFetchingAccountConversationsHeatmap) {
+        return;
+      }
+
+      // the data for the last 6 days won't ever change,
+      // so there's no need to fetch it again
+      // but we can write some logic to check if the data is already there
+      // if it is there, we can refetch data only for today all over again
+      // and reconcile it with the rest of the data
+      // this will reduce the load on the server doing number crunching
+      let to = endOfDay(new Date());
+      let from = startOfDay(subDays(to, 6));
+      let reconcile = false;
+
+      if (this.accountConversationHeatmap.length) {
+        to = endOfDay(new Date());
+        from = startOfDay(to);
+        reconcile = true;
+      }
+
+      this.$store.dispatch('fetchAccountConversationHeatmap', {
+        reportObj: {
+          metric: 'conversations_count',
+          from: getUnixTime(from),
+          to: getUnixTime(to),
+          groupBy: 'hour',
+          businessHours: false,
+        },
+        _options: {
+          reconcile,
+        },
+      });
```

Comment:
Is this over-engineering 👀 
CC: @pranavrajs @nithindavid

Reply:
@scmmishra I'm not sure if I understand this. Can you please explain a bit more on what you are trying to do?

Reply:
Sure, in the first case, if there's no data, the entire weeks data is fetched from the API. Since this method is called in the callback of an event, this can result in multiple API calls for this. This can be heavy for the server, so for subsequent API calls only fetch data for today. This is a significantly lighter query, the data is then reconciled with the existing data